packages/rlc-common/src/transformSampleGroups.ts

```diff
@@ -0,0 +1,420 @@
+  return [pathItself].concat(pathParams);
+}
+
+function convertMethodLevelParameters(
+  methods: OperationMethod[],
+  importedDict: Record<string, Set<string>>,
+  schemaMap: Map<string, Schema>,
+  packageName: string,
+  operationParameter?: OperationParameter
+): SampleParameter[] {
+  if (
+    !methods ||
+    methods.length === 0 ||
+    !operationParameter ||
+    operationParameter.parameters.length === 0
+  ) {
+    return [];
+  }
+  const rawMethodParams = operationParameter.parameters;
+  const method = methods[0];
+  const requestParameter = rawMethodParams[0];
+  const hasInputParams = !!rawMethodParams && rawMethodParams.length > 0,
+    requireParam = !method.hasOptionalOptions;
+  if (!hasInputParams && !requireParam) {
+    return [];
+  }
+
+  const allSideAssignments = [],
+    querySideAssignments: string[] = [],
+    headerSideAssignments: string[] = [];
+  if (
+    !!requestParameter.body &&
+    requestParameter.body &&
+    requestParameter.body.body &&
+    requestParameter.body.body?.length > 0
+  ) {
+    const body = requestParameter.body.body[0];
+    const bodyTypeName = body.typeName ?? body.type;
+    if (bodyTypeName !== "string" && body.oriSchema) {
+      schemaMap.set(bodyTypeName, body.oriSchema);
+    }
+    allSideAssignments.push(
+      ` body: ` + generateParameterTypeValue(bodyTypeName, "body", schemaMap)
+    );
+  }
+
+  requestParameter.parameters
+    ?.filter((p) => p.type === "query")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      querySideAssignments.push(
+        `${name}: ` + generateParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+
+  if (querySideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` queryParameters: { ` + querySideAssignments.join(", ") + `}`
+    );
+  }
+  requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() !== "contenttype")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      headerSideAssignments.push(
+        `${name}: ` + generateParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+  if (headerSideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` headers: { ` + headerSideAssignments.join(", ") + `}`
+    );
+  }
+  const contentType = requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() === "contenttype");
+  if (contentType && contentType.length > 0) {
+    allSideAssignments.push(
+      ` ${contentType[0].name}: ` +
+        generateParameterTypeValue(
+          contentType[0].param.type,
+          contentType[0].name,
+          schemaMap
+        )
+    );
+  }
+  let value: string = `{}`;
+  if (allSideAssignments.length > 0) {
+    value = `{ ` + allSideAssignments.join(", ") + `}`;
+  } else {
+    return [];
+  }
+
+  const optionParam: SampleParameter = {
+    name: "options",
+    assignment: `const options: ${method.optionsName} =` + value + `;`
```

Comment:
I don't think we should encourage customer to use type annotation in any rlc example.

Reply:
removed